components/Switch/__test__/switch.spec.ts

```diff
@@ -0,0 +1,201 @@
+import { tick } from 'svelte';
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+import KSwitch from '../src';
+
+let host: HTMLElement;
+
+const initHost = () => {
+	host = document.createElement('div');
+	host.setAttribute('id', 'host');
+	document.body.appendChild(host);
+};
+beforeEach(() => {
+	initHost();
+	vi.useFakeTimers();
+});
+afterEach(() => {
+	host.remove();
+	vi.restoreAllMocks();
+});
+
+describe('Test: KSwitch', () => {
+	test('props: value', async () => {
+		const value = true;
+		const instance = new KSwitch({
+			target: host,
+			props: {
+				value
+			}
+		});
+		expect(instance).toBeTruthy();
+		expect(host.innerHTML.includes('k-switch__checked')).toBe(true);
+		instance.$set({
+			value: false
+		});
+		await tick();
+		expect(host.innerHTML.includes('k-switch__un_checked')).toBe(true);
+		expect(host.innerHTML).matchSnapshot();
+	});
+
+	test('props: unCheckedValue and checkedValue', async () => {
+		let value = '微笑';
+		const instance = new KSwitch({
+			target: host,
+			props: {
+				value,
+				unCheckedValue: '哭泣',
+				checkedValue: '微笑'
+			}
+		});
+		expect(instance).toBeTruthy();
+		await tick();
+		instance.$on('updateValue', (v) => {
+			value = v.detail;
+		});
+		const switchElm = host.getElementsByTagName('div')[0];
+		switchElm.click();
+		await tick();
+		expect(value).toBe('哭泣');
+		switchElm.click();
+		await tick();
+		expect(value).toBe('微笑');
+		expect(host.innerHTML).matchSnapshot();
+	});
+
+	test('props: disabled', async () => {
+		const instance = new KSwitch({
+			target: host,
```

Comment:
The two props of disable and loading should also need to test whether the event is triggered and whether the value has changed

Reply:
thanks, I'll finish it as soon as possible